cmd/podman/cp.go

```diff
@@ -225,19 +220,38 @@ func copyBetweenHostAndContainer(runtime *libpod.Runtime, src string, dest strin
 		destPath = filepath.Join(dir, destPath)
 	}
 
-	var lastError error
-	for _, src := range glob {
-		if src == "-" {
-			src = os.Stdin.Name()
-			extract = true
-		}
+	var files []string
+	if srcPath[len(srcPath)-2:] == ALL_FILES {
+		base := srcPath[:len(srcPath)-1]
+		err = filepath.Walk(base, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err // prevent panic by handling failure accessing a path
+			}
+			if info.IsDir() {
+				if path == base { // Walk feeds us root first, skipping means zero files walked
+					return nil
+				}
+				return filepath.SkipDir // No recursive descents...
```

Comment:
Can’t this just be a `Readdir` or perhaps even `Readdirnames` instead of `Walk`, if it does not recurse, then? (I have no idea whether it would end up shorter; maybe not.)

Reply:
@mtrmac When I tried Walk it became longer and harder to read.

Reply:
The Readdir loop came out longer and more code since it required re-joining paths.  I reverted to Walk() after trying it.